test/tstDBSCAN.cpp

```diff
@@ -0,0 +1,171 @@
+/****************************************************************************
+ * Copyright (c) 2017-2021 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+#include "ArborX_DBSCANVerification.hpp"
+#include "ArborX_EnableDeviceTypes.hpp" // ARBORX_DEVICE_TYPES
+#include <ArborX_DBSCAN.hpp>
+
+#include <boost/test/unit_test.hpp>
+
+template <typename DeviceType, typename T>
+auto buildView(std::vector<T> const &v)
+{
+  using ExecutionSpace = typename DeviceType::execution_space;
+
+  Kokkos::View<T *, DeviceType> view("Testing::v", v.size());
+  Kokkos::deep_copy(ExecutionSpace{}, view,
+                    Kokkos::View<T const *, Kokkos::HostSpace,
+                                 Kokkos::MemoryTraits<Kokkos::Unmanaged>>(
+                        v.data(), v.size()));
```

Comment:
```suggestion
  Kokkos::deep_copy(view,
                    Kokkos::View<T const *, Kokkos::HostSpace,
                                 Kokkos::MemoryTraits<Kokkos::Unmanaged>>(
                        v.data(), v.size()));
```
Using the three-argument deep-copy here is a little pointless if we don't pass an execution space instance as argument. We are relying on the fact that we always get the same stream when calling the default constructor.